app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,58 @@
+class ProjectsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @projects = current_user.projects
+  end
+
+  def new
+    @project = current_user.projects.new
+  end
+
+  def create
+    @project = current_user.projects.new(project_params)
+
+    if @project.save
+      flash[:success] = "Project create"
+      @project.members.create(user_id: current_user.id)
+      redirect_to @project
+    else
+      flash.now[:danger] = "Project don\'t create"
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show
+    @project = Project.find(params[:id])
```

Comment:
You can move `@project = Project.find(params[:id])` to the `before_action` callback. This code will be executed before the action.
```
before_action :find_project, only: [:show, :edit, :update, :destroy] #add any actions you need

...

def find_project
  @project = Project.find(params[:id])
end
```

Reply:
https://rusrails.ru/action-controller-overview#kolbeki-ekshna

Reply:
Done